spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -620,6 +627,7 @@ def add_electrodes(recording: se.RecordingExtractor, nwbfile=None, metadata: dic
                     description=descr
                 )
 
+    # TODO: do not save gain and offset when write_scaled is True!
```

Comment:
@CodyCBakerPhD can you look into this?

Reply:
So, as I look at it especially with respect to the treatment of things like Intan whose return_unscaled is by default now uint16, I'll also need to re-fix the `write_scaled=False` case, too.

Reply:
I think that is working! But when `write_scaled` is Tru gain and offset should not be stored in the electrode table

Reply:
Again, we really would prefer to avoid writing the offset at all. Ignore I even mentioned the previous solution to that. The `add_electrical_series` should operate on signed int types, or floats. If something returns an unsigned type, we should correct it using the scaled offset.

Reply:
That's exactly what the gain and offset do right now. return_scaled is True by default, but at least this gives the opportunity to save raw raw data.

Reply:
OK help me out here - I'm walking through the logic of getting traces from an IntanRecordingExtractor: https://github.com/SpikeInterface/spikeextractors/blob/6ebc754511632ac0d134775dbc9b1fb9231888b5/spikeextractors/extractors/intanrecordingextractor/intanrecordingextractor.py#L80-L130

I want my traces to return as `int16`, with `recording.get_channel_gains` returning the conversion factors for each channel. This is the best mapping to the Nwb ElectricalSeries.

But as I look at the decorator: https://github.com/SpikeInterface/spikeextractors/blob/6ebc754511632ac0d134775dbc9b1fb9231888b5/spikeextractors/extraction_tools.py#L762-L807

I can only set `return_scaled` to `True` (returns series as microvolt floats) or `False` (returns the `uint16`).

Reply:
So, with this update intan can return either uint16 (raw) or float (scaled). The int16 version was a partial conversion that we made because we didn't have the offset mechanism on place

Reply:
Right, because we didn't have the offset mechanism in place on spikeextractors or in the NWB schema. You have it in spikeextractors now but the NWB side will take longer. So I'm saying for the add_electrical_series, I'll have to do that partial conversion thing as the default behavior when `write_scaled` is `False` and `recording.has_unscaled` is `True`. 

Because otherwise we can't tell without trying to infer the dtype, and we could end up writing certain types of raw data to the acquisition without all of the information to naturally convert it to volts.